chapter_11_cqrs.asciidoc

```diff
@@ -31,25 +31,32 @@ git checkout chapter_10_external_events
 ----
 ====
 
-=== Post/Redirect/Get
 
-If you've been doing web development, you're probably familiar with the
-post/redirect/get pattern. This is a technique where a web endpoint accepts an
-HTTP POST, and responds with a redirect to see the result. For example, we might
-accept a POST to `/batches` to create a new batch, and redirect the user to
-`/batches/123` to see their newly created batch.
+=== Async Post  and Post/Redirect/Get
 
-This approach fixes the problems that arise when users refresh the results page
+If you've used REST APIs, you're probably familiar with the
+Async Post or Post/Redirct/Get patterns.
+
+Async Post is a technique where a HTTP endpoint accepts an
+HTTP POST, and responds with a 202 Accepted, with resource link to see the result.
+For example, we might accept a POST to `/batches` to create a new batch, and
+return a link to `/batches/123`, where the client can retrieve their newly created batch.
+
+Async Post fixes the problem where a POST may trigger a long running operation,
+and we do not want to block the user while waiting on the result
+
+When used together with a Redirect, it's called Post/Redirect/Get.
+Post/Redirect/Get fixes the problems that arise when users refresh the results page
```

Comment:
excellent nitpick as usual.  i'm tempted to invert the order of explanations and do PRG first and mention the async post variant second.  probably because i came from a django / traditional web-dev background, so (with no further evidence) assume everyone else does too.  if nothing else it's chronological i suppose.

Reply:
hmm.  now that i think about it, are we really doing an async post?  the 202 status implies processing has not completed, but actually we are pretty much applying all the state changes synchronously before returning a response....

Reply:
Had to pick this nit, since I was a bit thrown off by the mention of PRG since there was never any re-direct., and I've been bit recently by some of these things.

You can effectively treat a sync operation as an async operation that occurs instantly, so `202 should not be a problem.

`201` should also be sensible, since the allocations have been created synchronously.  Then, there are two options for the API: (1) the client has tribal knowledge/API docs and knows to do a `GET` on `allocations/<order_id>` or `allocations?order_id=<order_id>`. (2) The client relies on the server embedding the route in the 201 response body, and just does a `GET` on that.  Then there's no re-direct at all and you can delete the PRG discussion.